File: 922 - Sort Array By Parity II.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        int even = 0, odd = 1, n = nums.size();
        while (even < n && odd < n) {
            while (even < n && nums[even] % 2 == 0) even += 2;
            while (odd < n && nums[odd] % 2 == 1) odd += 2;
            if (even < n && odd < n) {
                int t = nums[even];
                nums[even] = nums[odd];
                nums[odd] = t;
            }
            
        }
        return nums;
    }
};

File: 922 - Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, nums: list[int]) -> list[int]:
        odd, even = 0, 1
        n = len(nums)
        while odd < n or even < n:
            while odd < n and nums[odd] % 2 == 0:
                odd += 2
            while even < n and nums[even] % 2 == 1:
                even += 2
            if odd < n and even < n:
                nums[odd], nums[even] = nums[even], nums[odd]
        return nums